### PR TITLE
feat: cyclonedx parser

### LINKED ISSUE #623: new SBOM parser
There is a concept called SBOM (Software Bill Of Materials), and there are various tooling around this[^1][^2][^3][^4] concept to generate it. There are several methods designed specifically for delivering SBOMs, including [SPDX (Software Package Data Exchange)](https://spdx.dev), [Software Identification (SWID) Tags](https://nvd.nist.gov/products/swid), and [Cyclone DX](https://cyclonedx.org).

Wouldn't be nice if we support validating these types of SBOM specs via rego-based policies in `conftest`.

[^1]: https://github.com/anchore/syft
[^2]:  https://github.com/spdx/spdx-sbom-generator
[^3]: https://github.com/tern-tools/tern
[^4]: https://github.com/kubernetes/release/tree/master/cmd/bom

cc: @luhring @nishakm @puerco 🥳🙋🏻‍♂️

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>
Co-authored-by: Furkan Türkal <furkan.turkal@trendyol.com>

Fixes #623 
ezgif com-gif-maker(6)

cc: @jalseth @dentrax

### CODE REVIEW

**Comment by developer-guy** [comment]
kindly ping @jalseth

**Comment by jpreese** [comment]
@developer-guy thanks for the PR. This seems straight-forward to me. Couple of thoughts:

- Would it be possible to use a smaller test fixture for this functionality? 11,000+ lines is quite.. bulky, and I think a much much smaller example, even if a little fabricated, may be a bit more digestable.
- Could some tests be added for this?

Thanks!

**Comment by boranx** [comment]
+1 to adding unit tests, also we might consider having an example with an appropriate policy under [examples/](https://github.com/open-policy-agent/conftest/tree/master/examples) that could be shown in the documentation.

**Comment by developer-guy** [comment]
thank you so much, folks @boranx @jpreese. I've updated the according to your requests.

Screen Shot 2021-11-26 at 10 53 22

**Review by boranx — CHANGES_REQUESTED** [review]
nice! I've left 2 more minor thing 
other than those, would you like to include your test to [acceptance.bats](https://github.com/open-policy-agent/conftest/blob/master/acceptance.bats) to make sure the functionality is ok always

Thanks for picking up this&for the contributions :+1:

**Comment by jalseth** [comment]
@developer-guy As an aside, please copy/paste errors, terminal text, etc. into the comments rather than using screenshots as we want to be inclusive of everyone including those that need to rely on screen readers.

File: report.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (developer-guy) [reply report.json]: I added them unintentionally. I created those just for local testing. I removed them right now. Thank you.

File: bom.json [hunk 2]
```diff
@@ -0,0 +1,11273 @@
+{
```
[the remark this answers is not in the document]
Comment (developer-guy) [reply bom.json:1]: same, I removed this one too.

**Comment by Dentrax** [comment]
Hey @boranx! Just updated to c3003d02e5edcc2ddfc7bfdb68947abd6cc4f5f3, this commit adds new `XML` support for `cyclonedx`, which is [built-in supported](https://github.com/CycloneDX/cyclonedx-go/blob/e0d515e31f467886d1d5370247d34160ab56b3e9/cyclonedx.go#L70-L75).

**Review by boranx — APPROVED** [review]
/lgtm thanks!

**Pull request merged**